### PR TITLE
DOCSP-32300 Adds compatibility info to landing page

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-32300
Staging - https://docs-mongodborg-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-32300/
Build - https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=6512fd66492df295cff4a9c5

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/index.txt [hunk 1]
```diff
@@ -46,6 +46,13 @@ If your Java application requires asynchronous stream processing, use the
 :driver:`Reactive Streams Driver </reactive-streams/>` which uses Reactive
 Streams to make non-blocking calls to MongoDB.
 
+Compatibility
+-------------
+
+You can use the Java driver to connect to deployments hosted in the
```
Comment (mongoKart) [inline source/index.txt:52]:
```suggestion
You can use the {+driver-short+} to connect to deployments hosted in the
```

**Review by mongoKart — CHANGES_REQUESTED** [review]
1 change

**Review by mongoKart — APPROVED** [review]
lgtm

**Review by kpytlar — COMMENTED** [review]
LGTM

**Pull request merged**